app/home/routes.py

```diff
@@ -57,38 +77,126 @@ def get_segment(request):
+        return redirect(url_for('home_blueprint.account'))
+
+    elif request.method == 'GET':
+        form.username.data = current_user.username
+        form.email.data = current_user.email
+    # get the path for the profile picture of the current user
+    profile_picture = url_for('static', filename=f'{current_user.profile_image}')
+
+    property_by_user = Property.query.filter_by(user_id=current_user.id).order_by(Property.date.desc())
+    photos = [json.loads(p.photos) for p in property_by_user]
+
+    return render_template('account.html',
+                           form=form,
+                           property_by_user=property_by_user,
+                           profile_picture=profile_picture,
+                           photos=photos)
+
+
+@blueprint.route('/update_property/<int:property_id>', methods=['GET', 'POST'])
+@login_required
+def update_property(property_id):
+    prop_to_update = Property.query.get_or_404(property_id)
+    form = PropertyForm()
+
+    if form.validate_on_submit():
         img_files = request.files.getlist('prop_photos')
+        imgs_folder = prop_to_update.image_folder
 
-        # get the id of the recently added property by current_user and use
-        # it to link the image files to that property.
-        get_prop_id = db.session.query(Property).order_by(Property.date.desc())
+        image_list = {'photos': [imgs_folder]}
 
-        # saves image files uploaded to app/base/static/property_images/
-        for img_file in img_files:
-            filename = secure_filename(img_file.filename)
+        property_image_handler(img_files, image_list, imgs_folder)
+        image_list_to_json = json.dumps(image_list)
+
+        prop_to_update.name = form.prop_name.data
+        prop_to_update.desc = form.prop_desc.data
+        prop_to_update.price = form.prop_price.data
+        prop_to_update.photos = image_list_to_json
+        prop_to_update.location = form.prop_location.data
+        prop_to_update.date = datetime.utcnow()
+        db.session.commit()
 
-            # replace spaces in filename with underscore
-            clean_filename = filename.replace(' ', '_')
-            prop_images = PropertyImage(image_name=clean_filename,
-                                        property_id=get_prop_id[0].id)
-            db.session.add(prop_images)
-            db.session.commit()
+        flash("Your Property listing has been updated", 'success')
+        return redirect(url_for('home_blueprint.view_property'))
 
-            file_path = os.path.join(current_app.root_path, 'base/static/property_images', clean_filename)
-            img_file.save(file_path)
+    elif request.method == 'GET':
+        # prefills the forms with the attributes to the property to be updated
+        form.prop_name.data = prop_to_update.name
+        form.prop_desc.data = prop_to_update.desc
+        form.prop_price.data = prop_to_update.price
+        form.prop_location.data = prop_to_update.location
```

Comment:
some pseudocode to consider for optimisation here, just for your thoughts 😛 
```
for data in form.data:
   form.item = property_to_update.data.item
   ```

Reply:
I am getting an error  'PropertyForm' object has no attribute 'items' in the browser